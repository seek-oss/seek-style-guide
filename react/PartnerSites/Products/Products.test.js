import React from 'react';
import { render } from 'enzyme';

import Products from './Products';

const renderProducts = props => render(<Products {...props} />);
const linkRenderer = props => (<a {...props} />);

describe('Products:', () => {
  describe('locale states:', () => {
    it('should render with locale of AU', () => {
      expect(renderProducts({ locale: 'AU', linkRenderer })).toMatchSnapshot();
    });

    it('should render with locale of NZ', () => {
      expect(renderProducts({ locale: 'NZ', linkRenderer })).toMatchSnapshot();
    });
  });
  describe('active product states:', () => {
    it('should render with no link as active by default', () => {
      expect(renderProducts({ locale: 'AU', linkRenderer })).toMatchSnapshot();
    });
    it('should render with jobs link as active', () => {
      expect(renderProducts({ locale: 'AU', linkRenderer, activeProduct: 'Jobs' })).toMatchSnapshot();
    });
    it('should render with courses link as active', () => {
      expect(renderProducts({ locale: 'AU', linkRenderer, activeProduct: 'Courses' })).toMatchSnapshot();
    });
    it('should render with businesses link as active', () => {
      expect(renderProducts({ locale: 'AU', linkRenderer, activeProduct: 'Businesses for sale' })).toMatchSnapshot();
    });
    it('should render with volunteering link as active', () => {
      expect(renderProducts({ locale: 'AU', linkRenderer, activeProduct: 'Volunteering' })).toMatchSnapshot();
    });
  });
});
