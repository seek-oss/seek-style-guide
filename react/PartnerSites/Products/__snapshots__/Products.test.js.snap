// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products: active product states: should render with businesses link as active 1`] = `
<nav
  aria-labelledby="Products"
  role="navigation"
>
  <span
    class="root"
  >
    <h1
      id="Products"
    >
      SEEK Products
    </h1>
  </span>
  <ul
    class="list"
  >
    <li>
      <a
        class="link link_isJobs"
        data-analytics="header:jobs"
        href="https://www.seek.com.au"
        title="SEEK Jobs"
      >
        Jobs
      </a>
    </li>
    <li>
      <a
        class="link link_isLearning"
        data-analytics="header:courses"
        href="https://www.seek.com.au/learning/?campaigncode=seek_banner_29&sc_trk=skj-courses-link"
        title="SEEK Learning"
      >
        Courses
      </a>
    </li>
    <li>
      <span
        class="link link_isActive"
        title="SEEK Business"
      >
        Businesses for sale
      </span>
    </li>
    <li>
      <a
        class="link link_isVolunteering"
        data-analytics="header:volunteering"
        href="https://www.volunteer.com.au/?tracking=SKMAU:main+header"
        title="SEEK Volunteer"
      >
        Volunteering
      </a>
    </li>
  </ul>
</nav>
`;

exports[`Products: active product states: should render with courses link as active 1`] = `
<nav
  aria-labelledby="Products"
  role="navigation"
>
  <span
    class="root"
  >
    <h1
      id="Products"
    >
      SEEK Products
    </h1>
  </span>
  <ul
    class="list"
  >
    <li>
      <a
        class="link link_isJobs"
        data-analytics="header:jobs"
        href="https://www.seek.com.au"
        title="SEEK Jobs"
      >
        Jobs
      </a>
    </li>
    <li>
      <span
        class="link link_isActive"
        title="SEEK Learning"
      >
        Courses
      </span>
    </li>
    <li>
      <a
        class="link link_isBusiness"
        data-analytics="header:business+for+sale"
        href="https://www.seekbusiness.com.au/?tracking=sk:main:au:nav:bus"
        title="SEEK Business"
      >
        Businesses for sale
      </a>
    </li>
    <li>
      <a
        class="link link_isVolunteering"
        data-analytics="header:volunteering"
        href="https://www.volunteer.com.au/?tracking=SKMAU:main+header"
        title="SEEK Volunteer"
      >
        Volunteering
      </a>
    </li>
  </ul>
</nav>
`;

exports[`Products: active product states: should render with jobs link as active 1`] = `
<nav
  aria-labelledby="Products"
  role="navigation"
>
  <span
    class="root"
  >
    <h1
      id="Products"
    >
      SEEK Products
    </h1>
  </span>
  <ul
    class="list"
  >
    <li>
      <span
        class="link link_isActive"
        title="SEEK Jobs"
      >
        Jobs
      </span>
    </li>
    <li>
      <a
        class="link link_isLearning"
        data-analytics="header:courses"
        href="https://www.seek.com.au/learning/?campaigncode=seek_banner_29&sc_trk=skj-courses-link"
        title="SEEK Learning"
      >
        Courses
      </a>
    </li>
    <li>
      <a
        class="link link_isBusiness"
        data-analytics="header:business+for+sale"
        href="https://www.seekbusiness.com.au/?tracking=sk:main:au:nav:bus"
        title="SEEK Business"
      >
        Businesses for sale
      </a>
    </li>
    <li>
      <a
        class="link link_isVolunteering"
        data-analytics="header:volunteering"
        href="https://www.volunteer.com.au/?tracking=SKMAU:main+header"
        title="SEEK Volunteer"
      >
        Volunteering
      </a>
    </li>
  </ul>
</nav>
`;

exports[`Products: active product states: should render with no link as active by default 1`] = `
<nav
  aria-labelledby="Products"
  role="navigation"
>
  <span
    class="root"
  >
    <h1
      id="Products"
    >
      SEEK Products
    </h1>
  </span>
  <ul
    class="list"
  >
    <li>
      <a
        class="link link_isJobs"
        data-analytics="header:jobs"
        href="https://www.seek.com.au"
        title="SEEK Jobs"
      >
        Jobs
      </a>
    </li>
    <li>
      <a
        class="link link_isLearning"
        data-analytics="header:courses"
        href="https://www.seek.com.au/learning/?campaigncode=seek_banner_29&sc_trk=skj-courses-link"
        title="SEEK Learning"
      >
        Courses
      </a>
    </li>
    <li>
      <a
        class="link link_isBusiness"
        data-analytics="header:business+for+sale"
        href="https://www.seekbusiness.com.au/?tracking=sk:main:au:nav:bus"
        title="SEEK Business"
      >
        Businesses for sale
      </a>
    </li>
    <li>
      <a
        class="link link_isVolunteering"
        data-analytics="header:volunteering"
        href="https://www.volunteer.com.au/?tracking=SKMAU:main+header"
        title="SEEK Volunteer"
      >
        Volunteering
      </a>
    </li>
  </ul>
</nav>
`;

exports[`Products: active product states: should render with volunteering link as active 1`] = `
<nav
  aria-labelledby="Products"
  role="navigation"
>
  <span
    class="root"
  >
    <h1
      id="Products"
    >
      SEEK Products
    </h1>
  </span>
  <ul
    class="list"
  >
    <li>
      <a
        class="link link_isJobs"
        data-analytics="header:jobs"
        href="https://www.seek.com.au"
        title="SEEK Jobs"
      >
        Jobs
      </a>
    </li>
    <li>
      <a
        class="link link_isLearning"
        data-analytics="header:courses"
        href="https://www.seek.com.au/learning/?campaigncode=seek_banner_29&sc_trk=skj-courses-link"
        title="SEEK Learning"
      >
        Courses
      </a>
    </li>
    <li>
      <a
        class="link link_isBusiness"
        data-analytics="header:business+for+sale"
        href="https://www.seekbusiness.com.au/?tracking=sk:main:au:nav:bus"
        title="SEEK Business"
      >
        Businesses for sale
      </a>
    </li>
    <li>
      <span
        class="link link_isActive"
        title="SEEK Volunteer"
      >
        Volunteering
      </span>
    </li>
  </ul>
</nav>
`;

exports[`Products: locale states: should render with locale of AU 1`] = `
<nav
  aria-labelledby="Products"
  role="navigation"
>
  <span
    class="root"
  >
    <h1
      id="Products"
    >
      SEEK Products
    </h1>
  </span>
  <ul
    class="list"
  >
    <li>
      <a
        class="link link_isJobs"
        data-analytics="header:jobs"
        href="https://www.seek.com.au"
        title="SEEK Jobs"
      >
        Jobs
      </a>
    </li>
    <li>
      <a
        class="link link_isLearning"
        data-analytics="header:courses"
        href="https://www.seek.com.au/learning/?campaigncode=seek_banner_29&sc_trk=skj-courses-link"
        title="SEEK Learning"
      >
        Courses
      </a>
    </li>
    <li>
      <a
        class="link link_isBusiness"
        data-analytics="header:business+for+sale"
        href="https://www.seekbusiness.com.au/?tracking=sk:main:au:nav:bus"
        title="SEEK Business"
      >
        Businesses for sale
      </a>
    </li>
    <li>
      <a
        class="link link_isVolunteering"
        data-analytics="header:volunteering"
        href="https://www.volunteer.com.au/?tracking=SKMAU:main+header"
        title="SEEK Volunteer"
      >
        Volunteering
      </a>
    </li>
  </ul>
</nav>
`;

exports[`Products: locale states: should render with locale of NZ 1`] = `
<nav
  aria-labelledby="Products"
  role="navigation"
>
  <span
    class="root"
  >
    <h1
      id="Products"
    >
      SEEK Products
    </h1>
  </span>
  <ul
    class="list"
  >
    <li>
      <a
        class="link link_isJobs"
        data-analytics="header:jobs"
        href="https://www.seek.co.nz"
        title="SEEK Jobs"
      >
        Jobs
      </a>
    </li>
    <li>
      <a
        class="link link_isLearning"
        data-analytics="header:courses"
        href="https://www.seeklearning.co.nz/?campaigncode=seek_banner_29&sc_trk=skj-courses-link"
        title="SEEK Learning"
      >
        Courses
      </a>
    </li>
    <li>
      <a
        class="link link_isBusiness"
        data-analytics="header:business+for+sale"
        href="https://www.seekbusiness.com.au/?site=nz&tracking=sk:main:nz:nav:bus"
        title="SEEK Business"
      >
        Businesses for sale
      </a>
    </li>
    <li>
      <a
        class="link link_isVolunteering"
        data-analytics="header:volunteering"
        href="https://seekvolunteer.co.nz/?tracking=SKMNZ:main+header"
        title="SEEK Volunteer"
      >
        Volunteering
      </a>
    </li>
  </ul>
</nav>
`;
